edk2toolext/os/UefiVariableSupportLib.py

```diff
@@ -0,0 +1,310 @@
+                )
+                self._SetFirmwareEnvironmentVariable.restype = c_int
+                self._SetFirmwareEnvironmentVariable.argtypes = [
+                    c_wchar_p,
+                    c_wchar_p,
+                    c_void_p,
+                    c_int,
+                ]
+                self._SetFirmwareEnvironmentVariableEx = (
+                    windll.kernel32.SetFirmwareEnvironmentVariableExW
+                )
+                self._SetFirmwareEnvironmentVariableEx.restype = c_int
+                self._SetFirmwareEnvironmentVariableEx.argtypes = [
+                    c_wchar_p,
+                    c_wchar_p,
+                    c_void_p,
+                    c_int,
+                    c_int,
+                ]
+            except Exception:
+                logging.warn(
+                    "G[S]etFirmwareEnvironmentVariableW function doesn't seem to exist"
+                )
+                pass
+        else:
+            pass
+
+    #
+    # Function to get variable
+    # return a tuple of error code and variable data as string
+    #
+    def GetUefiVar(self, name, guid):
+        err = 0
+        if os.name == 'nt':
+            efi_var = create_string_buffer(EFI_VAR_MAX_BUFFER_SIZE)
+            if self._GetFirmwareEnvironmentVariable is not None:
+                logging.info(
+                    "calling GetFirmwareEnvironmentVariable( name='%s', GUID='%s' ).."
+                    % (name, "{%s}" % guid)
+                )
+                length = self._GetFirmwareEnvironmentVariable(
+                    name, "{%s}" % guid, efi_var, EFI_VAR_MAX_BUFFER_SIZE
+                )
+            if (0 == length) or (efi_var is None):
+                err = windll.kernel32.GetLastError()
+                if err != 0 and err != UefiVariable.ERROR_ENVVAR_NOT_FOUND:
+                    logging.error(
+                        "GetFirmwareEnvironmentVariable[Ex] failed (GetLastError = 0x%x)" % err
+                    )
+                    logging.error(WinError())
+                if efi_var is None:
+                    return (err, None)
+            return (err, efi_var[:length])
+
+        else:
+            # the variable name is VariableName-Guid
+            path = '/sys/firmware/efi/efivars/' + name + '-%s' % guid
+
+            if not os.path.exists(path):
+                err = UefiVariable.ERROR_ENVVAR_NOT_FOUND
+                return (err, None)
+
+            efi_var = create_string_buffer(EFI_VAR_MAX_BUFFER_SIZE)
+            with open(path, 'rb') as fd:
+                efi_var = fd.read()
+
+            return (err, efi_var[:length])
+
+    #
+    # Function to get all variable names
+    # return a tuple of error code and variable names byte array formatted as:
+    #
+    # typedef struct _VARIABLE_NAME {
+    #   ULONG NextEntryOffset;
+    #   GUID VendorGuid;
+    #   WCHAR Name[ANYSIZE_ARRAY];
+    # } VARIABLE_NAME, *PVARIABLE_NAME;
+    #
+    def GetUefiAllVarNames(self):
```

Comment:
As ruff has yelled at you already, we use a different format for function comments. You can see other modules :)

Reply:
should be addressed in latest.